proto/trace.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+package trace;
+
+// The informantion to trace a RPC call to a component.
+// Setting any field to 0 will disable tracing on the RPC callee.
+message TraceContext {
+    // The id that are able to identify an unique request. It's usually an UUID.
+    uint64 trace_id = 1;
+    // The span that represents the caller's RPC calling procedural.
+    // Set to 0 when reporting to indicate that it's the root component of the entire trace.
+    uint64 parent_id = 2;
+}
+
+// The tracing spans for a RPC call on a component.
+message TraceRecord {
+    TraceContext context = 1;
+    repeated Span spans = 2;
+}
+
+message Span {
+    // The unique span id within a `TraceRecord`.
+    uint32 span_id = 1;
+    // The parent span within a `TraceRecord`.
+    // Set to 0 to indicate that it's the root span within a `TraceRecord`.
+    uint32 parent_id = 2;
+    uint64 begin_unix_ns = 3;
+    uint64 duration_ns = 4;
+    string event = 5;
+    repeated Property properties = 6;
+}
+
+message Property {
+    string key = 1;
+    string value = 2;
+}
+
+service TraceRecordPubSub {
+    // Clients subscribe tracing records through this RPC, the subscribed component will periodically (e.g. per minute)
+    // publishes tracing records to clients via gRPC stream.
+    rpc Subscribe(TraceRecordRequest) returns (stream TraceRecord) {}
+}
```

Comment:
I have some worries for this part, as it may be not compatible with how TiDB actually use. The TiDB will expect to receive trace results through the response synchronously.

Reply:
Yes, all reporting will be asynchronous in this design.